avalanche/benchmarks/utils/utils.py

```diff
@@ -673,12 +674,293 @@ def _get_task_labels_field(self) -> DataAttribute[int]:
+    dataset,
+    targets,
+    check_shape=True
+):
+    """
+    Initializes the targets for a given dataset.
+
+    To support backwards compatibility for when when 
+    :func:`create_multi_dataset_generic_benchmark` was
+    used to manage classification benchmarks only, this function will try to
+    mimic the steps taken in :func:`make_classification_dataset`, that is:
+    
+    - will try to check if the input dataset has classification 
+        targets (integer tensors / ndarray) and will cast them to
+        a list of native ints, as expected by other parts
+        of Avalanche.
+    - accepts passing an int for the targets field. The given int
+        will be applied to all exemplars in the dataset. 
+    - supports PyTorch TensorDataset, by taking the second tensor as targets.
+
+    If targets are not of type int, then they will be returned as-is,
+    so that other types of datasets (regression, detection, ...) are
+    supported without issues.
+
+    :param dataset: The input dataset. If the `targets` parameter is
+        not None, then targets will be retrieved from the dataset.
+    :param targets: The targets to use. Can be None, in which case
+        targets will be retrieved from the dataset.
+    :param check_shape: If True, will check if the number of exemplars
+        in the dataset match the length of the obtained targets sequence.
+    :return: The targets, as a DataAttribute of elements whose type depends
+        on the input dataset.
+    """
+    if targets is not None:
+        # User defined targets always take precedence
+        if isinstance(targets, int):
+            # Classification targets
+            targets = ConstantSequence(targets, len(dataset))
+        elif len(targets) != len(dataset) and check_shape:
+            raise ValueError(
+                "Invalid number of target labels. It must be equal to the "
+                "number of patterns in the dataset. Got {}, expected "
+                "{}!".format(len(targets), len(dataset))
+            )
+        return DataAttribute(targets, "targets")
+
+    targets = _traverse_supported_dataset(
+        dataset, _smart_select_targets_opt)
+    
+    if targets is not None:
+        # Classification targets
+        targets = to_int_list(targets, force=False)
+
+    if targets is None:
+        return None
+    
+    return DataAttribute(targets, "targets")
+
+
+def _smart_select_targets_opt(
+        dataset: Any,
+        indices: Optional[List[int]]) -> Optional[Sequence[Any]]:
+    if hasattr(dataset, "targets"):
+        # Standard supported dataset
+        found_targets = dataset.targets
+    elif hasattr(dataset, "tensors") and len(dataset.tensors) >= 2:
+        # Support for PyTorch TensorDataset
+        found_targets = dataset.tensors[1]
+    else:
+        return None
+
+    if indices is not None:
+        found_targets = SubSequence(found_targets, indices=indices)
+
+    return found_targets
+
+
+def make_generic_dataset(
```

Comment:
do we need it? also, it's not generic because it's still asks for targets

Reply:
I think it's still useful to load targets and task labels if they are available. Targets and task labels are optional (it's a best effort search).